geo-traits/src/polygon.rs

```diff
@@ -0,0 +1,64 @@
+use super::line_string::LineStringTrait;
+use geo_types::{CoordNum, LineString, Polygon};
+use std::iter::Cloned;
+use std::slice::Iter;
+
+pub trait PolygonTrait<'a>: Send + Sync {
+    type ItemType: 'a + LineStringTrait<'a>;
```

Comment:
Maybe `RingType` would be more descriptive? 🤷🏻

Reply:
Should we change the name `ItemType` in each trait then? I.e. `CoordType` in `LineStringTrait`?